qualtran/bloqs/chemistry/trotter.py

```diff
@@ -0,0 +1,553 @@
+    """
+    # We compute the inverse square root of x^2 using variable spaced QROM, interpolation and newton-raphson
+    # Build data so QROM recognizes repeated entries so as to use the variable spaced QROM implementation.
+    # The repeated ranges occur for l :-> l + 2^k, and are repeated twice
+    # for coeffs_a and coeffs_b. We need to scale the coefficients by 2^{-(k-1)}
+    # to correclty account for the selection range (r_{ij}^2)
+    # Our coefficients are initially defined in the range [1, 3/2] for "_a" and [3/2, 2] for "_b".
+    poly_coeffs_a, poly_coeffs_b = get_polynomial_coeffs()
+    data = np.zeros((4, 2 ** (selection_bitsize)), dtype=np.int_)
+    for i, (a, b) in enumerate(zip(poly_coeffs_a, poly_coeffs_b)):
+        # x = 0 is never encountered as the minimum distance is 1 for our grid.
+        num_data = 0
+        # x = 1
+        _, coeff = float_as_fixed_width_int(a, target_bitsize)
+        data[i, 1] = coeff
+        # x = 2
+        _, coeff = float_as_fixed_width_int(b, target_bitsize)
+        data[i, 2] = coeff
+        # x = 3
+        _, coeff = float_as_fixed_width_int(a / 2 ** (1 / 2), target_bitsize)
+        data[i, 3] = coeff
+        start = 4
+        for k in range(2, selection_bitsize):
+            _, coeff = float_as_fixed_width_int(a / 2 ** (k / 2), target_bitsize)
+            # Number of time to repeat the data.
+            data_size = max(1, 2 ** (k - 1))
+            num_data += data_size
+            end = start + data_size
+            data[i, start:end] = coeff
+            _, coeff = float_as_fixed_width_int(b / 2 ** (k / 2), target_bitsize)
+            start += data_size
+            end += data_size
+            data[i, start:end] = coeff
+            start = end
+    return data
+
+
+@frozen
+class QuantumVariableRotation(Bloq):
+    r"""Bloq implementing Quantum Variable Rotation
+
+    $$
+        \sum_j c_j|\phi_j\rangle \rightarrow \sum_j e^{i \xi \phi_j}  c_j | \phi_j\rangle
+    $$
+
+    This is the basic implementation in Fig. 14 of the reference.
+
+    Args:
+        bitsize: The number of bits encoding the phase angle $\phi_j$.
+
+    Register:
+     - phi: a bitsize size register storing the angle $\phi_j$.
+
+    References:
+        (Faster quantum chemistry simulation on fault-tolerant quantum
+            computers)[https://iopscience.iop.org/article/10.1088/1367-2630/14/11/115023/meta]
+            Fig 14.
+    """
+    phi_bitsize: int
+
+    @cached_property
+    def signature(self) -> Signature:
+        return Signature([Register('phi', bitsize=self.phi_bitsize)])
+
+    def short_name(self) -> str:
+        return 'e^{i*phi}'
+
+    def t_complexity(self) -> 'TComplexity':
+        # Upper bounding for othe moment with just phi_bitsize * Rz rotation gates.
+        return self.phi_bitsize * Rz(0.0).t_complexity()
+
+    def bloq_counts(self, ssa: Optional['SympySymbolAllocator'] = None) -> Set[Tuple[int, Bloq]]:
```

Comment:
need type import for SympySymbolAllocator

Reply:
Done.